### ISSUE TITLE
Es6 class inheritance breaks draw (and others)

### ISSUE THREAD

**Issue by vypxl** [processing/p5.js#4215]
#### Nature of issue?

- [x] Found a bug
- [ ] Existing feature enhancement
- [x] New feature request

#### Most appropriate sub-area of p5.js?

- [ ] Color
- [x] Core/Environment/Rendering
- [ ] Data
- [ ] Events
- [ ] Image
- [ ] IO
- [ ] Math
- [ ] Typography
- [ ] Utilities
- [ ] WebGL
- [ ] Other (specify if possible)

#### Details about the bug / new feature details:

- p5.js version: v0.10.2
- Web browser and version: Google Chrome 79.0.3945.88 (Official Build) (64-bit)
- Operating System: Linux 5.4.5-arch1-1 x86_64
- Steps to reproduce this:
I tried to use instance mode without a seed function but with class inheritance:

```js
class Display extends p5 {
    constructor(parent) {
        super(() => {} /*empty seed method to prevent global mode*/, parent)
    }

    draw() {
        this.background(0)
        this.rect(10, 10, 100, 150)
    }

    setup() {
        this.createCanvas(512, 512)
        this.background(0)
    }
}
```

I derived this by looking into the source code [here](https://github.com/processing/p5.js/blob/master/src/core/main.js#L532), where the constructor calls the seed method, letting the seed define `setup`, `draw`, ... directly on the instance.
I then thought that my approach should lead to the same result.

**However** the unexpected result is, that in `draw`, `this` is not defined. `setup` on the other hand, works perfectly.

I managed to create a workaround by defining the broken metods in the constructor like so:

```js
...
constructor() { 
    ...
   this.draw = () => {
        this.background(0)
        this.rect(10, 10, 100, 150)
    }
}
```

I also tested a few other methods:
`mouseWheel`, `mouseDragged` worked fine when defined in the class body.
`keyPressed` did not work (`this` was undefined like in `draw`).

I think that allowing to subclass the p5 class, especially due to itself being implemented as a class in source, would be a more intuitive way to implement instance mode than the seed function.

#### tl;dr
When inheriting `p5` and defining the p5 methods like `draw` in the class body, some methods do not have `this` defined when called by p5 while others do.

**Comment by simranaggarwal1999** [comment]
Hey can I work upon this issue ?

**Comment by limzykenneth** [comment]
Class inheritence syntax for the p5 object was not something we specifically designed for and tested, thus at the moment it is considered not supported.

**Comment by Qianqianye** [comment]
I will close this issue for organizational purpose. Please feel free to reopen it if you'd like another round of discussion. Thank you!
